packages/mui-material/src/StepIcon/StepIcon.d.ts

```diff
@@ -1,11 +1,12 @@
 import * as React from 'react';
 import { SxProps } from '@mui/system';
-import { InternalStandardProps as StandardProps } from '..';
+import { InternalStandardProps as StandardProps, SvgIconOwnProps } from '..';
 import { Theme } from '../styles';
 import { StepIconClasses } from './stepIconClasses';
 
 export interface StepIconProps
-  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, 'children'> {
+  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, 'children'>,
+    Omit<SvgIconOwnProps, 'color' | 'children'> {
```

Comment:
All the props which are passed to `StepIcon` except `StepIconProps` are passed to `StepIconRoot` which is inherited from `SvgIcon`. so extending `SvgIconOwnProps` here seemed obvious

https://github.com/mui/material-ui/blob/412dcbf9d54b29d85353f1ff9947a78beb6ed7c1/packages/mui-material/src/StepIcon/StepIcon.js#L79-L85

Reply:
I think we should use `SvgIconProps` instead of `SvgIconOwnProps`, or is there a reason to use the later?

What is the `color` conflict?

Reply:
>I think we should use SvgIconProps instead of SvgIconOwnProps, or is there a reason to use the later?

Reason i went for `SvgIconOwnProps` instead of `SvgIconProps` is due to conflicting event handler types. For example type of onClick event handler is different in `React.HTMLAttributes<HTMLDivElement>` and `SvgIconProps`, since both are different types, typescript is unable to extend both types. Attached playground which explains the issue better

https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgMqQA4EkEHsQAKUuGAzshAB6QgAm5AShImAHQASAKgLIAyAgmDBRgAIwCukUgB4ufACLAAbgFEANhAC2EcAD4ANMiYsOPAUJESp01ADUA4nfvqtOsLuQBvAFDeAvkA

>What is the `color` conflict?

type of `color` in `React.HTMLAttributes<HTMLDivElement>` is `string | undefined` where as type of color in ` color?: OverridableStringUnion<
    | 'inherit'
    | 'action'
    | 'disabled'
    | 'primary'
    | 'secondary'
    | 'error'
    | 'info'
    | 'success'
    | 'warning',
    SvgIconPropsColorOverrides
  >;` due to this conflict i've omitted `color` type

Now that i think about this, since `color` prop passed to `StepIcon` gets passed to `SvgIcon` i think we should omit `color` from `React.HTMLAttributes<HTMLDivElement>` not from `SvgIconOwnProps`. what do you think? If we make `color` type more strict, would it be breaking change for users who didn't typed properply?

Reply:
> Reason i went for SvgIconOwnProps instead of SvgIconProps is due to conflicting event handler types.

I see. And is `React.HTMLAttributes<HTMLDivElement>` used correctly here? Shouldn't it be `React.HTMLAttributes<SVGSVGElement>`? The root is an SVG, right?

> type of color in React.HTMLAttributes<HTMLDivElement> is string | undefined

Same question that the one above. Did `string` work? Or do only the `SvgIconOwnProps.color` values work?

Reply:
>Shouldn't it be React.HTMLAttributes<SVGSVGElement>?

Initially i thought of doing this, but thought it would be breaking change for users if they depend on  `React.HTMLAttributes<HTMLDivElement>`

Anyway, i updated types here https://github.com/mui/material-ui/pull/44337/commits/463efe7c40bfef0b07caf28c13d9edded46cc88b https://github.com/mui/material-ui/pull/44337/commits/11a180766bb81ae81826883e68573b4b8f240c1f